tools/extra/qpafilter/qpafilter.py

```diff
@@ -107,80 +112,99 @@ def write_sensor(self, sensor):
         upper_warning_id = self.threshold_map['Upper Warning']
         upper_fatal_id = self.threshold_map['Upper Fatal']
 
-        self.outfile.write(struct.pack(BLOB_STRUCT_FORMAT,
+        self.payload.write(struct.pack(BLOB_STRUCT_FORMAT,
                                        sensor_id,
                                        upper_warning_id,
                                        warning))
-        self.outfile.write(struct.pack(BLOB_STRUCT_FORMAT,
+        self.payload.write(struct.pack(BLOB_STRUCT_FORMAT,
                                        sensor_id,
                                        upper_fatal_id,
                                        fatal))
 
 
-class blob_reader:
+class blob_reader_v0:
```

Comment:
Will we  be having other blob reader versions in the same file in the future?

Reply:
The version is for the format or spec of the binary file. If the format ever needs to change, we can use the  new version when we create the file with that changed spec. Any reader (software or hardware) can parse it according to that versioned spec.

Reply:
Yes, I'm trying to establish a pattern that would allow for multiple readers to exist in the same script.  I figured that, if the writer needs to be updated (next blob version), then a new version of the writer could just replace the previous writer; but at the same time, there's no reason why the same script couldn't dump various versions of the blob.